core/src/lib/code-generation/custom-snippet.ts

```diff
@@ -104,40 +139,99 @@ export function initCustomSnippet(): CustomSnippets {
  * @class CustomSnippets
  */
 export class CustomSnippets {
-    private allCustomSnippet: ICustomSnippet[] = [];
+    private getterFunctionCustomSnippet: ICustomSnippet[] = [];
+    private fillFormCustomSnippet: ICustomSnippet[] = [];
+    private clearFormCustomSnippet: ICustomSnippet[] = [];
+    /**
+     * adds a new CustomSnippet to the list of CustomSnippets, which will be used.
+     *
+     * @param {ICustomSnippet} customSnippet
+     * @memberof CustomSnippets
+     */
+    public addForGetterFunctions(customSnippet: ICustomSnippet): void {
+        this.getterFunctionCustomSnippet.push(customSnippet);
+    }
     /**
      * adds a new CustomSnippet to the list of CustomSnippets, which will be used.
      *
      * @param {ICustomSnippet} customSnippet
      * @memberof CustomSnippets
      */
-    public add(customSnippet: ICustomSnippet): void {
-        this.allCustomSnippet.push(customSnippet);
+    public addForFillForm(customSnippet: Required<ICustomSnippet>): void {
+        this.fillFormCustomSnippet.push(customSnippet);
+    }
+    /**
+     * adds a new CustomSnippet to the list of CustomSnippets, which will be used.
+     *
+     * @param {ICustomSnippet} customSnippet
+     * @memberof CustomSnippets
+     */
+    public addForClearForm(customSnippet: ICustomSnippet): void {
+        this.clearFormCustomSnippet.push(customSnippet);
     }
     /**
      * @private
      */
     public execute(
         element: E2eElement,
         codeBuilder: QueuedCodeBuilder,
-        protractorImports: string[],
         options: IPageObjectBuilderOptions,
+        snippetsFor: 'getterFunction' | 'fillForm' | 'clearForm',
```

Comment:
Enum?

Reply:
it does not make a big difference, because the type is not reused, this option is much simpler...